File: 1849-maximum-absolute-sum-of-any-subarray/maximum-absolute-sum-of-any-subarray.py
class Solution:
    def maxAbsoluteSum(self, nums: List[int]) -> int:
        maxRes = 0
        minRes = 0

        for num in nums:
            maxRes = max(0, maxRes + num)
            minRes = min(0, minRes + num)
        
        return maxRes - minRes

File: 1849-maximum-absolute-sum-of-any-subarray/maximum-absolute-sum-of-any-subarray.java
class Solution {
    public int maxAbsoluteSum(int[] nums) {
        int max_sum = 0;
        int min_sum = 0;

        for (int num : nums) {
            max_sum = Math.max(0, max_sum + num);
            min_sum = Math.min(0, min_sum + num);
        }

        return max_sum - min_sum;
    }
}